app/controllers/services/service_instances_controller.rb

```diff
@@ -360,7 +367,7 @@ def serialize(controller, space, opts, orphans=nil)
 
     def create_paginated_collection_renderer(service_instance)
       VCAP::CloudController::RestController::PaginatedCollectionRenderer.new(
-        VCAP::CloudController::RestController::SecureEagerLoader.new,
+        ServiceInstanceSharedToEagerLoader.new,
```

Comment:
This feels like a pretty fundamental change to service instances controller. Subverting the framework and how it expects to load associations could lead to confusing behavior later. I'm surprised this didn't break any existing tests.

Is there another way we could configure the framework to eager load organizations in addition to it's usual machinations?

Reply:
The `create_paginated_collection_renderer` method, the `ServiceInstanceSharedToEagerLoader` and the `ServiceInstanceSharedToSerializer` are only used by the '/v2/service_instances/:guid/shared_to' endpoint. These are used to support paginated rendering using the custom `ServiceInstanceSharedToPresenter` presenter. We could not use the existing serialiser class as SharedTo does not have an associated model object. The eager loader is overridden in order to load the organisation model that is required by the presenter to display the org name.

These changes are confined to the `shared_to` endpoint, so it hasn't changed behaviour elsewhere in the controller.